tensorbay/opendataset/SegTrack2/loader.py

```diff
@@ -0,0 +1,156 @@
+
+    """
+    root_path = os.path.join(os.path.abspath(os.path.expanduser(path)))
+    image_dir = os.path.join(root_path, "JPEGImages")
+    original_mask_dir = os.path.join(root_path, "GroundTruth")
+    mask_dir = os.path.join(root_path, "Masks")
+    dataset = Dataset(DATASET_NAME)
+    dataset.notes.is_continuous = True
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+    _load_segments(image_dir, original_mask_dir, mask_dir, dataset)
+    return dataset
+
+
+def _load_segments(image_dir: str, original_mask_dir: str, mask_dir: str, dataset: Dataset) -> None:
+    for instance_num, segment_names in _SEGMENTS_INFO.items():
+        for segment_name in segment_names:
+            segment = dataset.create_segment(segment_name)
+            original_mask_subdir = os.path.join(original_mask_dir, segment_name)
+            mask_subdir = os.path.join(mask_dir, segment_name)
+            os.makedirs(mask_subdir, exist_ok=True)
+            if segment_name in ("penguin", "monkeydog"):
+                filename_reformatter = _FILENAME_REFORMATTERS[1]
+            else:
+                filename_reformatter = _FILENAME_REFORMATTERS[0]
+            if segment_name == "cheetah":
+                get_mask: _MASK_GETTER = _get_cheetah_instance_mask
+            else:
+                get_mask = _get_instance_mask
+            for image_path in glob(os.path.join(os.path.join(image_dir, segment_name), "*")):
+                data = Data(image_path)
+                data.label.instance_mask = get_mask(
+                    image_path,
+                    mask_subdir,
+                    original_mask_subdir,
+                    instance_num,
+                    filename_reformatter,
+                )
+                segment.append(data)
+
+
+def _get_instance_mask(
+    image_path: str,
+    mask_subdir: str,
+    original_mask_subdir: str,
+    instance_num: int,
+    filename_reformatter: Callable[[str], str],
+) -> InstanceMask:
+    filename = filename_reformatter(os.path.basename(image_path))
+    mask_path = os.path.join(mask_subdir, f"{os.path.splitext(filename)[0]}.png")
+    if instance_num == 1:
+        mask = _get_reformatted_mask(os.path.join(original_mask_subdir, filename))
+    else:
+        mask = _get_reformatted_mask(os.path.join(original_mask_subdir, "1", filename))
+        for instance_id in range(2, instance_num + 1):
+            alter_mask = np.array(
+                Image.open(os.path.join(original_mask_subdir, str(instance_id), filename)),
+            )[:, :, 0]
+            mask[alter_mask == 255] = instance_id
+
+    Image.fromarray(mask).save(mask_path)
+    return InstanceMask(mask_path)
+
+
+def _get_cheetah_instance_mask(
+    image_path: str, mask_subdir: str, original_mask_subdir: str, _: int, __: Callable[[str], str]
+) -> InstanceMask:
+    filename = os.path.basename(image_path)
+    new_filename = f"{os.path.splitext(filename)[0]}.png"
```

Comment:
why not add "cheetah" into ("penguin", "monkeydog"), then we could use filename_reformatter

Reply:
The extensions of two groups of masks for "cheetah" are different, the first is `bmp` and the second is `png`